src/annotator/bucket-bar-client.js

```diff
@@ -65,8 +65,8 @@ export class BucketBarClient {
 
     this._updatePending = true;
     requestAnimationFrame(() => {
-      // In document with many annotations computing the anchor positions can
-      // block the JS event loop for up to 200 ms.
+      // Computing anchor positions may be expensive when there are many
+      // annotations or a forced layout reflow
```

Comment:
Computing anchor positions can be expensive if it triggers a layout reflow. If no reflows are required then this would be fast even if there were a lot of annotations. Depending on how the browser does reflows, the cost might be spread out over the `getBoundingClientRect` calls or might be paid all at once in the first `getBoundingClientRect` call.

Reply:
My observation in Chrome is that this could be expensive even when no reflows are involved.

In the following video, I enable the browser extension on a previously loaded document: (1) Anchors are added and (2) the console outputs the time that takes to compute the anchors positions for the new anchors.

https://user-images.githubusercontent.com/8555781/151801287-875721b8-f12e-4785-b601-3a4d0f1602b8.mp4

I can see that in one case it takes 53 ms.

Reply:
Based on our conversation in [Slack](https://hypothes-is.slack.com/archives/C1M8NH76X/p1643637320348609?thread_ts=1643386945.652579&cid=C1M8NH76X) I decided to remove the comment. I realised I don't fully understand this well enough what the browsing is doing.